### ISSUE TITLE
Order Locations in Create App Service and Other Similar Dialogs

### ISSUE THREAD

**Issue by selvasingh** [Microsoft/azure-tools-for-java#201]
Open Eclipse with Azure Toolkit for Eclipse
Sign in
Create a Web App
Start Publish to Azure
Press 'Create'
![image](https://cloud.githubusercontent.com/assets/3642444/24055900/1c0727ea-0aff-11e7-88ca-05736902613e.png)

**Comment by vladimir-shcherbakov** [comment]
if I sort locations - they won't be grouped by countries.

**Comment by martinsawicki** [comment]
yes, because of the peculiar way Azure names its regions. This issue is also why we named these things in a more logical way in the SDK `Region` enum -- we used COUNTRY_AREA pattern consistently, even though it was inconsistent with Azure's use. 

But here, we're calling the Locations API from Azure, so we have no other choice but to return the list of names as returned by Azure. Not great, but I don't see a better alternative, short of manually ordering them ourselves. But if we did that, then that will not look like an ordering to the user either.

The ideal solution would be to show the names differently, just like our Region enum, consistently: "Country Area". But that would mean we'd have to hard-code that and also look more different from Azure portal, which shows them in alphabetical order even though not by country:
![region](https://cloud.githubusercontent.com/assets/881767/24225905/56cba84a-0f20-11e7-858b-648eeaf2fb7d.png)

So I agreed this does not look great, but the other alternatives seem worse for other reasons, and at least the plugin's region listing would be consistent with the portal.
